cmd/fetchbranches/SemaphoreCI.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Semaphore environment variables https://semaphoreci.com/docs/available-environment-variables.html
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type SemaphoreCI struct{}
+
+// GetPR retrieves the PR number from the Semaphore environment
+func (ci *SemaphoreCI) GetPR() (int, error) {
+	val := os.Getenv("PULL_REQUEST_NUMBER")
+
+	if val == "" {
+		return -1, fmt.Errorf("pull_request_number environment variable is empty")
+	}
+
+	return strconv.Atoi(val)
+}
+
+// GetRepository retrieves the owner and the repository name from the Semaphore environment
+func (ci *SemaphoreCI) GetRepository() (string, string, error) {
+	semaphoreRepository := os.Getenv("SEMAPHORE_REPO_SLUG")
+
+	// Checks that repository name is not empty
+	if semaphoreRepository == "" {
+		return "", "", fmt.Errorf("semaphore_repo_slug environment variable is empty")
+	}
+
+	testRepository := strings.Split(semaphoreRepository, "/")
+
+	// Valid names should have and owner and a repository name
+	if len(testRepository) != 2 {
+		return "", "", fmt.Errorf("this is not a valid repository name")
+	}
+
+	ownerName := testRepository[0]
+	repositoryName := testRepository[1]
+
+	// Checks that the owner name is not empty
+	if ownerName == "" {
+		return "", "", fmt.Errorf("missing an owner name")
```

Comment:
Again, I'd include the value of `semaphoreRepository` in the error to help with debugging the issue.